c/driver/postgresql/postgres_copy_reader.h

```diff
@@ -1290,7 +1337,28 @@ class PostgresCopyStreamWriter {
       }
       const ArrowType arrow_type = schema_view.type;
       PostgresCopyFieldWriter* child_writer;
-      NANOARROW_RETURN_NOT_OK(MakeCopyFieldWriter(arrow_type, &child_writer, error));
+      if (arrow_type == NANOARROW_TYPE_TIMESTAMP) {
```

Comment:
This is pretty verbose but when trying to do something like

```cpp
enum ArrowTimeUnit unit = schema_view.time_unit
child_writer = new PostgresCopyTimestampFieldWriter<unit>();
```

I was getting `the value of ‘unit’ is not usable in a constant expression`

That makes sense but at the same time there may be a more succinct way of getting around that limitation instead of the switch / case here